### PR TITLE
Make the List Blob Azure work regardless if the endpoint has a trailing slash or not

### PR DESCRIPTION
**Pull request by primeroz**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

when the endpoint for azure disk is defined with a trailing slash it causes an exception 

```
Code: 49. DB::Exception: Expected prefix 'system-tables/mergetree//' in blob name 'system-tables/mergetree/__meta/aucicmbxpvoyblnggzcwvxsuneyzonpu/prefix.path'. (LOGICAL_ERROR), Stack trace (when copying this message, always include the lines below):
```

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [x]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [x]  Allow: Azure
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ ListBlobsPagedResponse ContainerClientWrapper::ListBlobs(const ListBlobsOptions
     new_options.Prefix = blob_prefix / options.Prefix.ValueOr("");
 
     auto response = client.ListBlobs(new_options);
-    auto blob_prefix_str = blob_prefix.empty() ? "" : blob_prefix.string() + "/";
+    auto blob_prefix_str = blob_prefix.empty() ? "" : (blob_prefix.string() + (blob_prefix.string().back() == '/' ? "" : "/"));
```
Comment (CurtizJ) [inline src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp:111]: I think we can use `fs::path(blob_prefix) / ""` here. It handles all three cases: https://godbolt.org/z/hT8sMMPEd.

File: src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp [hunk 2]
```diff
@@ -108,7 +109,7 @@ ListBlobsPagedResponse ContainerClientWrapper::ListBlobs(const ListBlobsOptions
     new_options.Prefix = blob_prefix / options.Prefix.ValueOr("");
 
     auto response = client.ListBlobs(new_options);
-    auto blob_prefix_str = blob_prefix.empty() ? "" : blob_prefix.string() + "/";
+    auto blob_prefix_str = fs::path(blob_prefix / "").string();
```
Comment (CurtizJ) [inline src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp:112]:
```suggestion
    String blob_prefix_str = blob_prefix / "";
```

Comment (CurtizJ) [reply src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp]: I've noticed that `blob_prefix` is already a `fs::path`.

Comment (primeroz) [reply src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp]: so we don't even need ot call `fs::path` at all ?

**Review by CurtizJ — APPROVED** [review]

**Comment by jkartseva** [comment]
I mistakenly assigned tags for the backports. Since we don't backport https://github.com/ClickHouse/ClickHouse/pull/65691, they don't make sense.

**Pull request merged**